src/recipe/recipe.service.ts

```diff
@@ -1,9 +1,23 @@
 import { Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { Model } from 'mongoose';
+import { PaginatedQueryDto } from '../dto/queries.dto';
+import { RecipeDto } from './dto/recipe.dto';
 import { Recipe, RecipeDocument } from './recipe.schema';
 
 @Injectable()
 export class RecipeService {
   constructor(@InjectModel(Recipe.name) private recipeModel: Model<RecipeDocument>) {}
+
+  searchInTitle(query: string, paginatedQueryDto?: PaginatedQueryDto): Promise<RecipeDto[]> {
+    const page = paginatedQueryDto?.page || 1;
+    const limit = paginatedQueryDto?.limit || 10;
+    return this.recipeModel
+      .find({ title: { $regex: query, $options: 'i' } })
```

Comment:
I'm not sure if passing user input as regex is safe, I'd make sure it doesn't contain any special characters first